Address issue with setting up the partial cache in benchmark

seddonym opened the pull request:
Prior to this, we would set up the state for the first iteration, but future iterations would be able to use the fully-populated cache.

Peter554 reviewed (approved):
A bit awkward, but seems a good idea.

I'm a bit confused that the benchmark seems to take the same time as fully from the cache - I would have expected it to be a bit slower. I suppose 2 and 15 are not very big numbers, maybe it makes sense to add a case where we are missing 100, or 1000 cache entries.

seddonym commented:
> I'm a bit confused that the benchmark seems to take the same time as fully from the cache - I would have expected it to be a bit slower. I suppose 2 and 15 are not very big numbers, maybe it makes sense to add a case where we are missing 100, or 1000 cache entries.

I've added one for 350 misses (about half of Django). It does slow the benchmarking job down quite a bit unfortunately, maybe we can remove it later.

I've also adjusted it so it copies a real module from Django instead of a one-liner, so the parser will have more work to do.
